app/controllers/user_registrations_controller.rb

```diff
@@ -6,6 +6,7 @@ def create
     @user = build_resource(params[:spree_user])
     if resource.save
       session[:spree_user_signup] = true
+      session[:confirmation_return_url] = params[:return_url]
```

Comment:
I guess we can't pass this info as parameters in the URL right? I'm always nervous using session for this kind of things.

Reply:
We can capture the return URL here, but then the user model later enqueues a job to create the email via `UserConfirmationsController`, but I think we can't directly pass variables to it at this point. Do you have any suggestions?

Reply:
Can't we pass URL parameters to `after_sign_in_path_for(@user)`?

Reply:
this instance of `after_sign_up_path_for` directs to a path when the user initially signs up, not when the email link is clicked. At this point we can capture the URL we want to redirect to, but we can't pass it directly to the model, as this will be triggered later, once the link is followed. It looks like Devise uses sessions by default to store these redirect URLs between events.